CMakeLists.txt

```diff
@@ -660,18 +652,16 @@ if (BUILD_TESTING)
                 -fsanitize=fuzzer -lstdc++
             )
 
-            # Set the output directory for the fuzzing binaries
-            set(FUZZ_BIN_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz")
-            set_target_properties(${TEST_NAME} PROPERTIES
-                RUNTIME_OUTPUT_DIRECTORY ${FUZZ_BIN_DIR}
-            )
-
             add_test(NAME ${TEST_NAME}
-                    COMMAND ${CMAKE_COMMAND} -E env
-                        DYLD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}/lib:${CMAKE_CURRENT_BINARY_DIR}/tests/testlib:${CMAKE_CURRENT_SOURCE_DIR}/libcrypto-root/lib:$ENV{DYLD_LIBRARY_PATH}
-                        LD_LIBRARY_PATH=${CMAKE_CURRENT_BINARY_DIR}/lib:${CMAKE_CURRENT_BINARY_DIR}/tests/testlib:${CMAKE_CURRENT_SOURCE_DIR}/libcrypto-root/lib:$ENV{LD_LIBRARY_PATH}
-                        bash ${SCRIPT_PATH} ${TEST_NAME} ${FUZZ_TIMEOUT_SEC} ${CORPUS_UPLOAD_LOC} ${ARTIFACT_UPLOAD_LOC}
-                    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz)
+                COMMAND ${CMAKE_COMMAND} -E env
+                    bash ${SCRIPT_PATH} 
+                    ${TEST_NAME} 
+                    ${FUZZ_TIMEOUT_SEC} 
+                    ${BUILD_DIR_PATH} 
+                    ${CORPUS_UPLOAD_LOC} 
```

Comment:
Not in scope for this PR: But I realized while reading this that I don't fully understand how our corpus stuff works. Some of the corpus is uploaded/downloaded from S3, but some of it is checked into the git repo? What are the differences? How do those interact?

If you already know the answer I think it'd be super worthwhile to document it at https://github.com/aws/s2n-tls/tree/main/tests/fuzz#corpus

If not could you open an issue to improve the documentation around that?

Reply:
This is next task on my tracking issue here: https://github.com/aws/s2n-tls/issues/4748#:~:text=Update%20usage%20guide%20for%20fuzz%20test%20in%20readme

To answer your question, we only download/upload the corpus from S3 if the `CORPUS_UPLOAD_LOC` environment variable is set. When errors occur during fuzz tests, logs are uploaded to `ARTIFACT_UPLOAD_LOC`. These variables are only set for the nightly scheduled job, as unzipping and zipping corpus files add some overhead to the test time. It’s also an expensive operation to trigger for every PR commit, and PRs shouldn't be the place to look for new bugs through fuzzing. Therefore, we reserve this process for the scheduled fuzz job, to iteratively accumulate and improve our corpus files over time.